### ISSUE TITLE
`no-unused-vars` fails for symbol properties used in object literals or accessed with dot notation

### ISSUE THREAD

**Issue by ben-page** [eslint/eslint#10909]
**Tell us about your environment**

* **ESLint Version:** v5.6.1
* **Node Version:** v10.11.0
* **npm Version:** yarn v1.9.4

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    'parserOptions': {
        'ecmaVersion': 2018,
        'ecmaFeature': {
            'impliedStrict': true
        }
    },
    'env': {
        'node': true,
        'es6': true
    },
    'extends': [
        'eslint:recommended'
    ]
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* eslint-disable no-console */

const $prop = Symbol('prop');

const obj = {
    $prop: 'property value 1' //symbol is used here
};

console.log(obj.$prop); //symbol is used again
```

```bash
eslint test.js
```

**What did you expect to happen?**
I expected `no-unused-vars` to recognize that `$prop` is used. 

**What actually happened? Please include the actual, raw output from ESLint.**
There are two lines that should have satisfied the `no-unused-vars` rule. But it's thrown anyway.
```
C:\XXX\test.js
  4:7  error  '$prop' is assigned a value but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by platinumazure** [comment]
Hi @ben-page, thanks for the issue.

I'm not sure this is working how you think it is. I'm pretty sure that `obj.$prop` uses the string property (i.e., is equivalent to `obj["$prop"]`).

You might want to try using computed property accesses to be sure:

```js
const $prop = Symbol('prop');

const obj = {
    [$prop]: 'property value 1' //symbol is used here
};

console.log(obj[$prop]); //symbol is used again
```

And you could check whether you're assigning to the symbol correctly by passing it in directly to the computed property access:

```js
console.log(obj.$prop === obj[Symbol('prop')]);    // Thinking this is false
```

Also, in general, dot notation property accesses do not access a variable with the same name as the property.

Hope this helps!

**Comment by ben-page** [comment]
Thanks, you are right.
